packages/vuetify/src/components/VProgressLinear/VProgressLinear.sass

```diff
@@ -136,46 +107,44 @@
 .v-progress-linear--fixed
   position: fixed
 
-.v-progress-linear--reactive
-  .v-progress-linear__content
-    pointer-events: none
-
 .v-progress-linear--rounded
-  border-radius: $progress-linear-border-radius
+  @include rounded($progress-linear-border-radius)
+
+  &.v-progress-linear--rounded-bar
+    .v-progress-linear__determinate,
+    .v-progress-linear__indeterminate
+      border-radius: inherit
 
 .v-progress-linear--striped
   .v-progress-linear__determinate
     background-image: $progress-linear-stripe-gradient
     background-size: $progress-linear-stripe-background-size
     background-repeat: repeat
 
-.v-progress-linear--query
-  .v-progress-linear__indeterminate--active
-    .long
-      animation-name: query-ltr
-      animation-duration: 2s
-      animation-iteration-count: infinite
-
-    .short
-      animation-name: query-short-ltr
-      animation-duration: 2s
-      animation-iteration-count: infinite
-
-  &.v-progress-linear--reverse
-    .v-progress-linear__indeterminate--active
-      .long
-        animation-name: query-rtl
-
-      .short
-        animation-name: query-short-rtl
-
-.v-progress-linear--visible
-  .v-progress-linear__indeterminate--active
+.v-progress-linear--active
+  .v-progress-linear__indeterminate
     .long, .short
       animation-play-state: running
+
   .v-progress-linear__stream
     animation-play-state: running
 
+.v-progress-linear--rounded-bar
+  .v-progress-linear__determinate,
+  .v-progress-linear__indeterminate,
+  .v-progress-linear__stream + .v-progress-linear__background
+    @include rounded($progress-linear-border-radius)
+
+  .v-progress-linear__determinate
+    @include ltr()
+      border-top-left-radius: 0
+      border-bottom-left-radius: 0
+
+    @include rtl()
+      border-top-right-radius: 0
+      border-bottom-right-radius: 0
```

Comment:
It's because there are a few different behaviors when using `rounded`, `rounded-bar`, and `rounded rounded-bar`.

![image](https://user-images.githubusercontent.com/9064066/128430497-82cd71d6-e460-4e23-83b7-fe53bad2cac3.png)

Reply:
Oh, hm. I was imagining `rounded-bar` always being used together with `rounded`. What's the markup for the bottom three?